src/EnergyPlus/CMakeLists.txt

```diff
@@ -612,3 +612,49 @@ if(UNIX AND NOT APPLE)
   COMMAND ${CMAKE_COMMAND} -E create_symlink energyplus EnergyPlus
   COMMAND ${CMAKE_COMMAND} -E rename EnergyPlus "${CMAKE_BINARY_DIR}/Products/EnergyPlus") 
 endif()
+
+# if we are building a package, we can do additional operations, such as run doc scripts
+if(BUILD_PACKAGE)
+  
+  # Some docs are generated on the fly here, create a dir for the 'built' files
+  set( DOCS_OUT "${CMAKE_BINARY_DIR}/autodocs" )
+  file( MAKE_DIRECTORY "${DOCS_OUT}" )
+  
+  # the output variables listing
+  add_custom_command( 
```

Comment:
Understood.  I assume almost no one has BUILD_PACKAGE on unless they are actually building a package, but I see your point.  I don't see a trigger like PRE_BUILD that would cause this only when actually building a package.  Do you know of one?